projects/igniteui-angular/src/lib/grids/pivot-grid/pivot-grid.component.ts

```diff
@@ -2228,12 +2229,15 @@ export class IgxPivotGridComponent extends IgxGridBaseDirective implements OnIni
                 }
             case "date":
```

Comment:
I was wondering if maybe would be nice to also include strings that represent dates, but that might be not a good idea to assume.

Reply:
I was wondering about that too, but the auto-gen of columns also does not do that. It only generates date columns if the field has a Date object. So I left it like that for the pivot config as well. 
Technically there's an init event for the dimensions so if someone wants to mark a specific dimension a different type or enable/disable it before rendering, they could.